pkg/controller/migration/namespace_migration.go

```diff
@@ -291,6 +291,11 @@ func (m *CoreNamespaceMigration) Run(ctx context.Context, log logr.Logger) error
 		return fmt.Errorf("failed to migrate all nodes: %s", err.Error())
 	}
 	log.V(1).Info("Nodes migrated")
+	err := m.waitForOperatorCalicoNodeDaemonsetReady(ctx, log)
```

Comment:
Is this needed at this point? It seems unnecessary from your description and because it is included as part of the CleanupMigration function.

Reply:
Yes, so after all the calico-node pods are migrated to the `calico-system` namespace, the daemonset state could be in a transient state. 
Then as the next reconcile removes the node selector from the daemonset and it finds the daemonset in this state, the update won't respect the `updateStrategy`. This was our experience, we saw mass updates of the calico-node pods.

Although I agree that it is hard to differentiate. It is also sure that if we remove the labels from the nodes first then from the daemonset, that could result in the same.
We have tested this many times (100+), and this combintatiion was working the best so far - otherwise there are so frequent changes, and it results situations like reported.

I can do some re-runs if you want to double-check these statements.

Thanks,
Adam